lib/data_update_scripts/20211208074428_migrate_logo_svg_data.rb

```diff
@@ -0,0 +1,25 @@
+module DataUpdateScripts
+  class MigrateLogoSvgData
+    def run
+      return if ::Settings::General.try(:logo_svg).blank? ||
+        (::Settings::General.try(:original_logo).present? && ::Settings::General.try(:resized_logo).present?)
+
+      temp_svg = Tempfile.new(["logo", ".svg"])
+      temp_svg.write(::Settings::General.logo_svg)
+      temp_svg.close
```

Comment:
We create a temporary file with the SVG string

Reply:
The ["Good practices" sectiob](https://ruby-doc.org/stdlib-3.0.2/libdoc/tempfile/rdoc/Tempfile.html#class-Tempfile-label-Good+practices) of the `Tempfile` documentation encourages the following form instead:

```ruby
Tempfile.create("logo.svg") do |file|
   # ...do something with file... e.g.
  file.write(::Settings::General.logo_svg))
end
```

To quote the [docs](https://ruby-doc.org/stdlib-3.0.2/libdoc/tempfile/rdoc/Tempfile.html#method-c-open)

> The File object will be automatically closed and the temporary file is removed after the block terminates, releasing all resources that the block created. The call returns the value of the block.